raiden-cli/src/routes/channels.ts

```diff
@@ -1,21 +1,33 @@
 import { Router, Request, Response, NextFunction } from 'express';
 import { first, pluck } from 'rxjs/operators';
-import { ErrorCodes, RaidenError, isntNil } from 'raiden-ts';
-import { Cli } from '../types';
+import { ErrorCodes, isntNil, ChannelState } from 'raiden-ts';
+import { Cli, ApiChannelState } from '../types';
 import {
   isInvalidParameterError,
   isTransactionWouldFailError,
   validateOptionalAddressParameter,
 } from '../utils/validation';
 import { flattenChannelDictionary, transformSdkChannelFormatToApi } from '../utils/channels';
 
-function isConflictError(error: RaidenError): boolean {
+function isConflictError(error: Error): boolean {
   return (
     [ErrorCodes.RDN_UNKNOWN_TOKEN_NETWORK, ErrorCodes.CNL_INVALID_STATE].includes(error.message) ||
     isTransactionWouldFailError(error)
   );
 }
 
+function isInsuficientFundsError(error: { message: string; code?: string | number }): boolean {
+  return (
+    error.code === 'INSUFFICIENT_FUNDS' ||
+    [
+      ErrorCodes.DTA_INVALID_AMOUNT,
+      ErrorCodes.DTA_INVALID_DEPOSIT,
```

Comment:
Isn't this more a `Bad Request` error? If I provide an invalid value to deposit this error would be not correct. Because it can't know if I actually have enough funds. This function is used to determine if we should send a `402 - Payment Required` (official HTTP name) error. These error codes are actually part of the definition for `isInvalidParameterError()` (as all `DTA_*` codes are since this is their shared property). This means depending on how you order the if-else conditions at the route, this will result in a different response.

Reply:
Hmm, yes, I was not that concerned with the specifics of the error codes. Unfortunately, we don't have a specific check to fail early on the deposit if not enough funds, instead the transaction just fails on `estimateGas`. I'll remove the errors here and create an issue to fix it on the SDK.

Reply:
If you could link the issue here after creation would be great. Just close the conversation then.

Reply:
https://github.com/raiden-network/light-client/issues/1905
Already working on it.